protos/datamodel/costs.proto

```diff
@@ -0,0 +1,236 @@
+  // Cross-zone network transfer costs
+  CostMetrics crossZone = 1;
+
+  // Cross-region network transfer costs
+  CostMetrics crossRegion = 2;
+
+  // Internet egress costs
+  CostMetrics internet = 3;
+
+  // Network transfer bytes sent
+  float transferBytes = 4;
+
+  // Network bytes received
+  float receiveBytes = 5;
+}
+
+// GPUCostDetail represents GPU-specific cost information
+// Maps to OpenCost's detailed GPU allocation tracking
+message GPUCostDetail {
+  // Base GPU cost
+  ResourceCost resource = 1;
+
+  // GPU device information for attribution (matches GPUDevice fields)
+  string ID = 2;              // References GPUDevice.ID
+  string ModelName = 3;       // References GPUDevice.ModelName
+  int32 DeviceNumber = 4;     // References GPUDevice.DeviceNumber
+
+  // GPU sharing information (if applicable)
+  bool isShared = 5;
+  float sharePercentage = 6;
+
+  // Usage metrics for cost correlation (from DCGM)
+  float usageAvg = 7;
+  float usageMax = 8;
+}
+
+// StorageCostDetail represents persistent volume cost information
+message StorageCostDetail {
+  // Volume identifier
+  string volumeId = 1;
+
+  // Storage class
+  string storageClass = 2;
+
+  // Provider resource ID (for cloud storage)
+  string providerResourceId = 3;
+
+  // Storage cost
+  CostMetrics cost = 4;
+
+  // Storage usage in byte-hours
+  float byteHours = 5;
+}
+
+// LoadBalancerCostDetail represents load balancer cost information
+message LoadBalancerCostDetail {
+  // Load balancer identifier
+  string loadBalancerId = 1;
+
+  // Associated service name
+  string serviceName = 2;
+
+  // Load balancer IP address
+  string ipAddress = 3;
+
+  // Whether this is a private load balancer
+  bool isPrivate = 4;
+
+  // Load balancer cost
+  CostMetrics cost = 5;
+
+  // Hours the load balancer was active
+  float hours = 6;
+}
+
+// AllocationCost represents complete cost breakdown for a resource allocation
+// This is the main cost container that gets embedded in resource messages
+message AllocationCost {
+  // Window for this cost calculation (inherits from cluster window if not specified)
+  TimeWindow window = 1;
+
+  // Resource-specific costs
+  ResourceCost cpu = 2;
+  ResourceCost ram = 3;
+
+  // GPU costs (can have multiple GPUs)
+  repeated GPUCostDetail gpu = 4;
+
+  // Storage costs (persistent volumes)
+  repeated StorageCostDetail storage = 5;
+
+  // Network costs with detailed breakdown
+  DetailedNetworkCost network = 6;
+
+  // Load balancer costs
+  repeated LoadBalancerCostDetail loadBalancers = 7;
+
+  // Shared costs allocated to this resource
+  CostMetrics shared = 8;
+
+  // External cloud costs allocated to this resource
+  CostMetrics external = 9;
+
+  // Pre-computed total cost for performance (sum of all above)
+  CostMetrics total = 10;
+}
+
+// CostAttribution represents hierarchical cost attribution information
+// Maps to OpenCost's AllocationProperties for cost attribution
+message CostAttribution {
```

Comment:
Is there a reason we couldn't call this `ContainerProperties`? And could we just put these, flatly, on the `Container` message? I'm not sure we need a separate message.